crmsh/utils.py

```diff
@@ -2287,4 +2253,104 @@ def interface_choice():
     return [nic for nic in interface_list if nic != "lo"]
+
+
+class InterfaceInfo(object):
+    def __init__(self, ipv6=False, second_hb=False):
+        self.ipv6 = ipv6
+        self.nic_info_dict = {}
+        self.second_hb = second_hb
+
+    def get_interface_info(self):
+        cmd = "ip -{} -o addr show".format(6 if self.ipv6 else 4)
+        rc, out, err = get_stdout_stderr(cmd)
+        if rc != 0:
+            raise ValueError(err)
+
+        # format on each line will like:
+        # 2: enp1s0    inet 192.168.122.241/24 brd 192.168.122.255 scope global enp1s0\       valid_lft forever preferred_lft forever
+        for line in out.splitlines():
+            _, nic, _, ip_with_mask, *_ = line.split()
+            if nic == "lo":
+                continue
+            # one nic might configured multi IP addresses
+            if nic not in self.nic_info_dict:
+                self.nic_info_dict[nic] = []
+            self.nic_info_dict[nic].append(Interface(ip_with_mask))
+
+        if not self.nic_info_dict:
+            raise ValueError("No address configured")
+        if self.second_hb and len(self.nic_info_dict) == 1:
+            raise ValueError("Cannot configure second heartbeat line, since only one address available")
+
+    def get_default_ip_list(self, specify_nic_list=[]):
+        _ip_list = []
+        for nic in specify_nic_list:
+            if nic not in self.all_nic:
+                raise ValueError("Failed to detect IP address for {}".format(nic))
+            _ip_list.append(self.nic_info_dict[nic][0].ip)
+        if self.second_hb and len(specify_nic_list) == 1:
+            for nic in self.all_nic:
+                if nic not in specify_nic_list:
+                    _ip_list.append(self.nic_info_dict[nic][0].ip)
+                    break
+        return _ip_list
+
+    @property
+    def all_nic(self):
+        return list(self.nic_info_dict.keys())
+
+    @property
+    def interface_list(self):
+        _interface_list = []
+        for interface in self.nic_info_dict.values():
+            _interface_list.extend(interface)
+        return _interface_list
+
+    @property
+    def ip_list(self):
+        return [interface.ip for interface in self.interface_list]
+
+    @property
+    def ip_with_mask_list(self):
+        return [interface.ip_with_mask for interface in self.interface_list]
+
+    @property
+    def network_list(self):
+        return [interface.network for interface in self.interface_list]
+
+
+class Interface(object):
+    def __init__(self, ip_with_mask):
+        self.ip = ip_with_mask.split('/')[0]
+        self.ip_with_mask = ip_with_mask
+        self.network = get_network(ip_with_mask)
+
+
+def get_network(ip_with_mask):
+    """
+    Get IP address's network
+    IPv6 support
+    """
+    return str(ipaddress.ip_interface(ip_with_mask).network.network_address)
+
+
+class NoDefaultRouteError(Exception):
+    pass
+
+
+def get_default_nic_from_route():
```

Comment:
I would add this method to the `InterfacesInfo` as classmethod as it's related to the interfaces data